nodeJS/express/installing_postgresql.md

```diff
@@ -0,0 +1,293 @@
+<!-- markdownlint-disable MD024 TOP004 -->
+
+### Introduction
+
+Data persistence is integral to the vast majority of web applications. Data persistence is achieved through databases. Being able to understand how to structure, build, and query your own database are important skills for any full-stack developer to have.
+
+For example, consider The Odin Project. Where is your username stored? Where are project submissions stored? Where does this lesson content, the one you're reading right now, get stored? Yes, a database!
+
+We've chosen [PostgreSQL](https://www.postgresql.org/) as our database of choice for the curriculum. All SQL knowledge that you've learned in the earlier course should be transferable to PostgreSQL. We'll also learn new PostgreSQL specific things. Watch through this [fun and short introduction video to PostgreSQL by Fireship](https://www.youtube.com/watch?v=n2Fluyr3lbc).
+
+<div class="lesson-note lesson-note--critical" markdown="1">
+
+#### Prerequisite SQL knowledge
+
+Make sure you've completed the [SQL course](https://www.theodinproject.com/paths/full-stack-javascript/courses/databases).
+
+This lesson and all subsequent lessons will assume you understand SQL syntax and concepts.
+
+</div>
+
+In this lesson, we'll install PostgreSQL and in the next lesson we'll learn how to use it in our Express application.
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- How to install PostgreSQL.
+
+### Installing PostgreSQL
+
+Pick your operating system below to get the appropriate steps for installing PostgreSQL.
+
+<details markdown="block">
+
+<summary class="dropDown-header">Linux</summary>
```

Comment:
@MaoShizhong From your comment on the other PR, yeah the way we're doing these details/summary elements isn't great. Right now the heading structure from an AT perspective is essentially:

- Installing PostgresSQL
  - Step 1: Make sure the system is up to date
  - ....other headings....
  - Step 1: Make sure the system is up to date

Without any context of an OS anywhere. We could put this in as is as I'm not 100% sure how best to go about it yet. We could have the actual steps be visible at all times, then in each step have a details/summary per OS, so something like the following screenshot:

![image](https://github.com/TheOdinProject/curriculum/assets/70952936/e3086995-1427-4c13-84ec-23552b92f02c)

Since the content of the `summary` element can't be given a Markdown heading (we could use HTML `h3` etc, but then we need to allow that in the lint rules). But I'd have to stew on it a bit more probably.

Reply:
@thatblindgeye I straight up copy pasted the existing lesson in the Rails path in to a new file in this PR. If we do intend to make these changes, might I suggest handling it in a separate PR?

Reply:
@01zulfi Oh no this is definitely something we're doing in other places, I hope my comment didn't come off as saying this was an issue you introduced or anything.

But yeah, this is something that can be a followup since it is something that will affect other lessons and there's still some ruminating to do on the idea.

Reply:
All good 😆 

I'm suggesting to open a separate issue/PR because this issue is relevant to other installation-esque lessons as well. Anyway, I'm happy to update this PR if we do reach a consensus on how to fix this issue.

Reply:
I vote separate PR to handle both this and the Rails version together once decided on the best way to handle this.
Single headings each with OS-specific dropdowns does look appealing, though IIRC @thatblindgeye, I think you mentioned accessibility issues with the current actual method of doing the dropdown itself? Since if we can find an accessible and logical alternative that also conforms to the heading issue, we'd be wanting to update any other lesson that does things like this, like any of the earlier installations lessons

Reply:
Yeah this all can be a followup